climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1158 @@
+                    self.orig = np.delete(self.orig, evt_idx_merge)
+                    self._set_frequency()
+                    self.fraction = self.intensity.copy()
+                    self.fraction.data.fill(1.0)
+
+            else:
+                self.intensity = sparse.csr_matrix(np.amax(self.intensity, 0))
+                self.event_id = np.array([np.max(self.event_id)+1])
+                self.event_name = list(map(str, self.event_id))
+                self.date = [np.min(self.date)]
+                self.date_end = [np.max(self.date_end)]
+                self.orig = np.ones(1, bool)
+                self._set_frequency()
+                self.fraction = self.intensity.copy()
+                self.fraction.data.fill(1.0)
+            LOGGER.info('The merged event has event_id %s', self.event_id[-1])
+
+        else:
+            self.intensity = sparse.csr_matrix(np.amax(self.intensity, 0))
+            self.event_id = np.array([np.max(self.event_id)+1])
+            self.orig = np.zeros(1, bool)
+            self._set_frequency()
+            self.fraction = self.intensity.copy()
+            self.fraction.data.fill(1.0)
+
+    def summarize_fires_to_seasons(self, year_start=None, year_end=None,
+                                   hemisphere=None):
+        """ Summarize historic fires into fire seasons.
+
+        Parameters:
+            year_start (int, optional): start year; fires before that
+                are cut; no cut if not specified
+            year_end (int, optional): end year; fires after that are cut;
+                no cut if not specified
+            hemisphere (str, optional): 'SHS' or 'NHS' to define fire seasons
+        """
+        
+        # define hemisphere
+        if hemisphere is None:
+            if self.centroids.lat[0] > 0:
+                hemisphere = 'NHS'
+            elif self.centroids.lat[0] < 0:
+                hemisphere = 'SHS'
+
+        # define years
+        fire_years = np.array([date.fromordinal(i).year for i in self.date])
+        years = np.arange(np.min(fire_years), np.max(fire_years)+1)
+        if year_start is not None:
+            years = np.delete(years, np.argwhere(years < year_start))
+        if year_end is not None:
+            years = np.delete(years, np.argwhere(years > year_end))
+
+        # summarize to fire season
+        date_new = np.zeros(len(years), int)
+        date_end_new = np.zeros(len(years), int)
+        n_fires = np.zeros(len(years), int)
+        intensity_new = sparse.lil_matrix(np.zeros((len(years), len(self.centroids.lat))))
+
+        for i, year in enumerate(years):
+            if hemisphere == 'NHS':
+                StartTime = date.toordinal(date(year, 1, 1))
+                EndTime = date.toordinal(date(year+1, 1, 1))
+            elif hemisphere == 'SHS':
+                StartTime = date.toordinal(date(year, 7, 1))
+                EndTime = date.toordinal(date(year+1, 7, 1))
```

Comment:
This fire season definition looks different from the one in `set_hist_fire_seasons_FIRMS`.

Reply:
within set_hist_fire_seasons_FIRMS the function _select_fire_season is called. The methodology there is consistent